shapes/typeIndex.shacl

```diff
@@ -0,0 +1,28 @@
+PREFIX sh: <http://www.w3.org/ns/shacl#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+PREFIX solid: <http://www.w3.org/ns/solid/terms#>
+PREFIX : <#>
+
+:TypeIndexShape a sh:Shape ;
+    sh:scopeClass solid:ListedDocument, solid:UnlistedDocument, solid:TypeIndex .
+
+:TypeRegistration a sh:Shape ;
+    sh:scopeClass solid:TypeRegistration ;
+    sh:property [
+        sh:predicate solid:forClass ;
+        sh:nodeKind sh:IRI ;
+        sh:minCount 1 ;
+        sh:maxCount 1 ;
+    ] ;
+    sh:property [
+        sh:predicate solid:instanceContainer ;
+        sh:nodeKind sh:IRI ;
+        sh:minCount 0 ;
+        sh:maxCount 1 ;
+    ] ;
+    sh:property [
+        sh:predicate solid:instance ;
+        sh:nodeKind sh:IRI ;
+        sh:minCount 0 ;
+        sh:maxCount 1 ;
+    ] .
```

Comment:
I believe a TypeRegistration has to have exactly one of solid:instanceContainer or solid:instance, so this might be specified more strictly using a [disjunction constraint](https://www.w3.org/TR/shacl/#OrConstraintComponent).

```suggestion
    sh:or ([
        sh:path solid:instanceContainer ;
        sh:nodeKind sh:IRI ;
        sh:minCount 0 ;
        sh:maxCount 1 ;
    ] [
        sh:path solid:instance ;
        sh:nodeKind sh:IRI ;
        sh:minCount 0 ;
        sh:maxCount 1 ;
    ]) .    
```

Moreover, since they have exactly the same constraints, this can be summarized using [alternative paths](https://www.w3.org/TR/shacl/#property-path-alternative).

```suggestion
    sh:property [
        sh:path [ sh:alternativePath ( 
            solid:instance 
            solid:instanceContainer 
        ) ] ;
        sh:nodeKind sh:IRI ;
        sh:minCount 0 ;
        sh:maxCount 1 ;
    ] .    
```

Reply:
🤩 I learned something right there!